<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="Junyan Lu" />

<meta name="date" content="2021-10-05" />

<title>Pre-processing GDSC drug screen datasets</title>

<script src="site_libs/header-attrs-2.11/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/main/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>








<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.tab('show');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">DepInfeR</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">



<h1 class="title toc-ignore">Pre-processing GDSC drug screen datasets</h1>
<h4 class="author">Junyan Lu</h4>
<h4 class="date">2021-10-05</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span> workflowr <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2021-12-24
</p>
<p>
<strong>Checks:</strong> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 6 <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> 1
</p>
<p>
<strong>Knit directory:</strong> <code>DepInfeR/analysis/</code> <span class="glyphicon glyphicon-question-sign" aria-hidden="true" title="This is the local directory in which the code in this file was executed."> </span>
</p>
<p>
This reproducible <a href="https://rmarkdown.rstudio.com">R Markdown</a> analysis was created with <a
  href="https://github.com/workflowr/workflowr">workflowr</a> (version 1.7.0). The <em>Checks</em> tab describes the reproducibility checks that were applied when the results were created. The <em>Past versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguncommittedchanges"> <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> <strong>R Markdown file:</strong> uncommitted changes </a>
</p>
</div>
<div id="strongRMarkdownfilestronguncommittedchanges" class="panel-collapse collapse">
<div class="panel-body">
<p>The R Markdown is untracked by Git. To know which version of the R Markdown file created these results, you’ll want to first commit it to the Git repo. If you’re still working on the analysis, you can ignore this warning. When you’re finished, you can run <code>wflow_publish</code> to commit the R Markdown file and build the HTML.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the global environment can affect the analysis in your R Markdown file in unknown ways. For reproduciblity it’s best to always run the code in an empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20211005code"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Seed:</strong> <code>set.seed(20211005)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20211005code" class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20211005)</code> was run prior to running the code in the R Markdown file. Setting a seed ensures that any results that rely on randomness, e.g. subsampling or permutations, are reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Session information:</strong> recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be confident that you successfully produced the results during this run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrong43be8a7"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Repository version:</strong> 43be8a7 </a>
</p>
</div>
<div id="strongRepositoryversionstrong43be8a7" class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development and connecting the code version to the results is critical for reproducibility.
</p>
<p>
The results in this page were generated with repository version 43be8a7. See the <em>Past versions</em> tab to see a history of the changes made to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for the analysis have been committed to Git prior to generating the results (you can use <code>wflow_publish</code> or <code>wflow_git_commit</code>). workflowr only checks the R Markdown file, but you know if there are other scripts or data files that it depends on. Below is the status of the Git repository when the results were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .DS_Store
    Ignored:    .Rhistory
    Ignored:    .Rproj.user/
    Ignored:    analysis/.DS_Store
    Ignored:    analysis/.Rhistory
    Ignored:    analysis/analysis_RNAseq_cache/
    Ignored:    data/.DS_Store
    Ignored:    output/.DS_Store

Untracked files:
    Untracked:  analysis/analysis_EMBL2016.Rmd
    Untracked:  analysis/analysis_GDSC.Rmd
    Untracked:  analysis/analysis_RNAseq.Rmd
    Untracked:  analysis/analysis_beatAML.Rmd
    Untracked:  analysis/process_EMBL2016.Rmd
    Untracked:  analysis/process_GDSC.Rmd
    Untracked:  analysis/process_beatAML.Rmd
    Untracked:  analysis/process_kinobeads.Rmd
    Untracked:  code/utils.R
    Untracked:  data/BeatAML/
    Untracked:  data/EMBL2016/
    Untracked:  data/GDSC/
    Untracked:  data/Kinobeads/
    Untracked:  data/RNAseq/
    Untracked:  manuscript/
    Untracked:  output/BeatAML_result.RData
    Untracked:  output/EMBL_result.RData
    Untracked:  output/EMBL_resultSub.RData
    Untracked:  output/GDSC_result.RData
    Untracked:  output/allTargets.rds
    Untracked:  output/inputs_BeatAML.RData
    Untracked:  output/inputs_EMBL.RData
    Untracked:  output/inputs_GDSC.RData

Unstaged changes:
    Modified:   README.md
    Modified:   _workflowr.yml
    Modified:   analysis/_site.yml
    Deleted:    analysis/about.Rmd
    Modified:   analysis/index.Rmd
    Deleted:    analysis/license.Rmd
    Deleted:    output/README.md

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not included in this status report because it is ok for generated content to have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">
<p>
There are no past versions. Publish this analysis with <code>wflow_publish()</code> to start tracking its development.
</p>
<hr>
</div>
</div>
</div>
<div id="load-packages" class="section level1">
<h1>Load packages</h1>
<p>Packages</p>
<pre class="r"><code>library(depInfeR)
library(RColorBrewer)
library(pheatmap)
library(ggbeeswarm)
library(ggrepel)
library(tidyverse)
source(&quot;../code/utils.R&quot;)
knitr::opts_chunk$set(dev = c(&quot;png&quot;,&quot;pdf&quot;))</code></pre>
</div>
<div id="load-pre-processed-datasets" class="section level1">
<h1>Load pre-processed datasets</h1>
<pre class="r"><code>load(&quot;../output/inputs_GDSC.RData&quot;)</code></pre>
<div id="dimensions-of-input-matrices" class="section level2">
<h2>Dimensions of input matrices</h2>
<p>Drug-target</p>
<pre class="r"><code>dim(tarMat_GDSC)</code></pre>
<pre><code>[1]  66 118</code></pre>
<p>Drug-sample (viability matrix)</p>
<pre class="r"><code>dim(viabMat_GDSC)</code></pre>
<pre><code>[1]  66 126</code></pre>
</div>
</div>
<div id="multivariant-model-for-protein-dependence-prediction" class="section level1">
<h1>Multivariant model for protein dependence prediction</h1>
<p>Perform multivariant LASSO regression based on a drug-protein affinity matrix and a drug response matrix.<br />
This chunk can take a long time to run. Therefore we will save the result for later use to save time.</p>
<pre class="r"><code>result &lt;- runLASSOregression(TargetMatrix = tarMat_GDSC, ResponseMatrix = viabMat_GDSC, seed = 333)

#remove targets that were never selected
useTar &lt;- rowSums(result$coefMat) != 0
result$coefMat &lt;- result$coefMat[useTar,]


#save intermediate results
save(result, file = &quot;../output/GDSC_result.RData&quot;)</code></pre>
<p>Load the saved result</p>
<pre class="r"><code>load(&quot;../output/GDSC_result.RData&quot;)</code></pre>
<p>Number of selected targets</p>
<pre class="r"><code>nrow(result$coefMat)</code></pre>
<pre><code>[1] 9</code></pre>
</div>
<div id="heatmap-plots-of-protein-dependence-matrix" class="section level1">
<h1>Heatmap plots of protein dependence matrix</h1>
<p>The protein dependence matrix can be nicely visualized in a heatmap. High positive coefficients imply strong reliance of a certain sample on this protein for survival. Proteins with coefficients close to zero are less essential for the cell’s survival. Negative coefficients indicate that the viability phenotype benefits from inhibition of the protein.</p>
<div id="heatmap-of-protein-dependence-coefficients" class="section level2">
<h2>Heatmap of protein dependence coefficients</h2>
<pre class="r"><code>annoColor &lt;- list(H2O2 = c(`-1` = &quot;red&quot;, `0` = &quot;black&quot;, `1` = &quot;green&quot;),
                  IL.1 = c(`-1` = &quot;red&quot;, `0` = &quot;black&quot;, `1` = &quot;green&quot;),
                  JAK.STAT = c(`-1` = &quot;red&quot;, `0` = &quot;black&quot;, `1` = &quot;green&quot;),
                  MAPK.only = c(`-1` = &quot;red&quot;, `0` = &quot;black&quot;, `1` = &quot;green&quot;),
                  MAPK.PI3K = c(`-1` = &quot;red&quot;, `0` = &quot;black&quot;),
                  TLR = c( `-1` = &quot;red&quot;, `0` = &quot;black&quot;, `1` = &quot;green&quot;),
                  Wnt = c(`-1` = &quot;red&quot;, `0` = &quot;black&quot;, `1` = &quot;green&quot;),
                  VEGF = c(`-1` = &quot;red&quot;, `0` = &quot;black&quot;, `1` = &quot;green&quot;),
                  PI3K.only = c(`-1` = &quot;red&quot;, `0` = &quot;black&quot;, `1` = &quot;green&quot;),
                  TCGA.classification = c(ALL=&quot;#BC3C29FF&quot;,AML=&quot;#E18727FF&quot;,DLBC=&quot;#20854EFF&quot;,&quot;BRCAHer-&quot;=&quot;#0072B5FF&quot;,&#39;BRCAHer+&#39;=&quot;#7876B1FF&quot;),
                  ARID1A_mut = c(`1` = &quot;black&quot;,`0` = &quot;grey80&quot;),
                  EP300_mut = c(`1` = &quot;black&quot;,`0` = &quot;grey80&quot;),
                  PTEN_mut = c(`1` = &quot;black&quot;,`0` = &quot;grey80&quot;),
                  TP53_mut = c(`1` = &quot;black&quot;,`0` = &quot;grey80&quot;),
                  PIK3CA_mut = c(`1` = &quot;black&quot;,`0` = &quot;grey80&quot;),
                  BRCA2_mut = c(`1` = &quot;black&quot;,`0` = &quot;grey80&quot;),
                  BRCA1_mut = c(`1` = &quot;black&quot;,`0` = &quot;grey80&quot;),
                  CDH1_mut = c(`1` = &quot;black&quot;,`0` = &quot;grey80&quot;),
                  FBXW7_mut = c(`1` = &quot;black&quot;,`0` = &quot;grey80&quot;),
                  NRAS_mut = c(`1` = &quot;black&quot;,`0` = &quot;grey80&quot;),
                  ASXL1_mut = c(`1` = &quot;black&quot;,`0` = &quot;grey80&quot;),
                  MLL2_mut = c(`1` = &quot;black&quot;,`0` = &quot;grey80&quot;),
                  ABL1_trans = c(`1` = &quot;black&quot;,`0` = &quot;grey80&quot;),
                  missing_value_perc = c(`0` = &quot;white&quot;,`25` = &quot;red&quot;)
)

plotTab &lt;- result$coefMat
#Row normalization while keeping sign
plotTab_scaled &lt;- scale(t(plotTab), center = FALSE, scale= TRUE)
plotTab &lt;- t(plotTab_scaled)
levels(mutation_GDSC$TCGA.classification) &lt;- c(levels(mutation_GDSC$TCGA.classification), &quot;BRCAHer-&quot;)
mutation_GDSC$TCGA.classification[mutation_GDSC$TCGA.classification==&quot;BRCA&quot;] &lt;- &quot;BRCAHer-&quot;
mutation_GDSC$TCGA.classification &lt;- factor(mutation_GDSC$TCGA.classification, levels = c(&quot;ALL&quot;, &quot;AML&quot;, &quot;DLBC&quot;, &quot;BRCAHer-&quot;, &quot;BRCAHer+&quot;))
mutation_GDSC$missing_value_perc &lt;- NULL

pheatmap(plotTab, 
         color=colorRampPalette(rev(brewer.pal(n = 7, name = &quot;RdBu&quot;)), bias= 1.8)(100),
         annotation_col = mutation_GDSC,
         annotation_colors = annoColor,
         clustering_method = &quot;ward.D2&quot;, scale = &quot;none&quot;, 
         show_colnames = TRUE, main = &quot;&quot;, fontsize = 9, fontsize_row = 10, fontsize_col = 7)</code></pre>
<p><img src="figure/analysis_GDSC.Rmd/GDSC_coefHeatmap-1.png" width="1728" style="display: block; margin: auto;" /></p>
</div>
</div>
<div id="differential-dependence-on-proteins-associated-with-cancer-types-and-genotypes" class="section level1">
<h1>Differential dependence on proteins associated with cancer types and genotypes</h1>
<p>Prepare genomic background table</p>
<pre class="r"><code>cell_anno_final &lt;- mutation_GDSC %&gt;%
  #dplyr::select(-missing_value_perc) %&gt;% 
  dplyr::rename(cancer_type = TCGA.classification) %&gt;% 
  dplyr::filter(rownames(mutation_GDSC) %in% colnames(result$coefMat))
colnames(cell_anno_final) &lt;- str_remove_all(colnames(cell_anno_final),&quot;_mut&quot;)
colnames(cell_anno_final) &lt;- str_replace_all(colnames(cell_anno_final),&quot;_trans&quot;,&quot;_translocation&quot;)</code></pre>
<p>Association test between protein dependence and cancer type or mutational background</p>
<pre class="r"><code>testRes &lt;-  diffImportance(result$coefMat, cell_anno_final)</code></pre>
<p>Visualize protein associations with cancer type</p>
<pre class="r"><code>CancerType &lt;- testRes %&gt;% dplyr::filter(mutName == &quot;cancer_type&quot;) %&gt;% dplyr::filter(p.adj &lt; 0.05, FC &gt; 0.1)
plotTab &lt;- t(scale(t(result$coefMat))) %&gt;% data.frame() %&gt;%
  rownames_to_column(&quot;target&quot;) %&gt;% gather(key = &quot;CellLine&quot;, value = &quot;coef&quot;,-target) %&gt;% mutate(Cancer_Type = mutation_GDSC[CellLine,]$TCGA.classification) %&gt;%
  group_by(target, Cancer_Type) %&gt;% mutate(meanCoef = mean(coef)) %&gt;% arrange(meanCoef) %&gt;% ungroup() %&gt;%
  mutate(target = factor(target, levels = unique(target))) 


plotTab &lt;- plotTab %&gt;% dplyr::filter(target %in% CancerType$targetName)
plotTab$Cancer_Type &lt;- factor(plotTab$Cancer_Type, levels = c(&quot;ALL&quot;, &quot;AML&quot;,&quot;DLBC&quot;, &quot;BRCAHer-&quot;, &quot;BRCAHer+&quot;))

ggplot(plotTab, aes(x = target, y = coef, group=Cancer_Type)) +
 geom_jitter(
  aes(color = Cancer_Type), 
  position = position_jitterdodge(jitter.width = 0.2, dodge.width = 0.8),
  size = 1.2
  ) +
  stat_summary(
    fun= mean, fun.min=mean, fun.max=mean, colour=&quot;grey25&quot;,
    geom = &quot;crossbar&quot;,  size = 0.8,
    position = position_dodge(0.8)
    ) +
  scale_color_manual(values= c(&quot;#BC3C29FF&quot;,&quot;#E18727FF&quot;,&quot;#20854EFF&quot;,&quot;#0072B5FF&quot;,&quot;#7876B1FF&quot;),
                     guide = guide_legend(override.aes = list(size = 3) )) +
  ggtitle(&quot;Protein dependence associated with cancer type&quot;) + ylab(&quot;Protein dependence coefficient&quot;) + xlab(&quot;Protein&quot;) + theme_bw() + 
  theme_custom + geom_vline(xintercept =seq(from = 1.5, to = 8.5, by = 1), color=&quot;darkgrey&quot;) + labs(color = &quot;Cancer Type&quot;)</code></pre>
<p><img src="figure/analysis_GDSC.Rmd/GDSC_diffCoef_box-1.png" width="1536" style="display: block; margin: auto;" /></p>
<p>Radar plot visualization</p>
<pre class="r"><code>starMatrix &lt;- plotTab  %&gt;% dplyr::select(target, Cancer_Type, meanCoef)  %&gt;% distinct()
starMatrix &lt;- starMatrix %&gt;% pivot_wider(names_from = target, values_from = meanCoef) 
starMatrix &lt;- starMatrix %&gt;% mutate(Cancer_Type =  factor(Cancer_Type, levels = c(&quot;ALL&quot;, &quot;AML&quot;,&quot;DLBC&quot;, &quot;BRCAHer-&quot;, &quot;BRCAHer+&quot;))) %&gt;% arrange(Cancer_Type) 
starMatrix &lt;- starMatrix %&gt;% column_to_rownames(&quot;Cancer_Type&quot;)
starMatrix_norm &lt;- (as.matrix(starMatrix) + abs(min(starMatrix)))
zeroValue &lt;-abs(min(starMatrix)) / max(starMatrix_norm)
starMatrix_norm &lt;- starMatrix_norm/ max(starMatrix_norm)


#function for a single star plot
starPlot &lt;- function(dataIn, sampleName,zeroVal, color=&#39;red&#39;) {
    stopifnot(is.matrix(dataIn) &amp; nrow(dataIn) == 1)
    
    #data for outer ring
    outer &lt;- dataIn
    outer[!is.na(outer)] &lt;- 1
    
    #data for inner ring
    inner &lt;- dataIn
    inner[!is.na(inner)] &lt;- zeroVal
    
    #plotting
    stars(outer, draw.segments = FALSE, scale=FALSE, full=TRUE, locations=c(1,1), mar = c(4,4,4,4), main=sampleName, cex=0.5) #plot the outter ring
    stars(inner, draw.segments = FALSE, scale=FALSE, full=TRUE, locations=c(1,1), lty =2 ,add=TRUE) #plot the inner ring
    stars(dataIn, col.stars=color, draw.segments = FALSE, scale=FALSE, full=TRUE, key.loc = c(1,1), key.labels = colnames(dataIn), location=c(1,1), add=TRUE,cex=1) #plot the actual data
}

starColor &lt;- c(&quot;#BC3C29FF&quot;,&quot;#E18727FF&quot;,&quot;#20854EFF&quot;,&quot;#0072B5FF&quot;,&quot;#7876B1FF&quot;) #define color scheme

par(mfrow=c(2,3))  #layout in 3 X 3 format
for (i in seq(1,nrow(starMatrix_norm))) {
  dataIn &lt;- starMatrix_norm[i, ,drop=FALSE]
  sampleName &lt;- rownames(starMatrix_norm)[i]
  starCol &lt;- starColor[i]
  starPlot(dataIn,sampleName,zeroValue, starCol)
}  </code></pre>
<p><img src="figure/analysis_GDSC.Rmd/GDSC_diffCoef_radar-1.png" width="624" style="display: block; margin: auto;" /></p>
<p>Visualize significant associations between protein dependence and mutational background</p>
<pre class="r"><code>colList2 &lt;- c(`not significant` = &quot;grey80&quot;, mutated = &quot;#BC3C29FF&quot;, unmutated = &quot;#0072B5FF&quot;)

pos = position_jitter(width = 0.15, seed = 10)

plotTab &lt;- testRes %&gt;% dplyr::filter(mutName != &quot;cancer_type&quot;) %&gt;% mutate(type = ifelse(p.adj &gt; 0.1, &quot;not significant&quot;, 
                                          ifelse(FC &gt;0, &quot;mutated&quot;,&quot;unmutated&quot;))) %&gt;%
  mutate(varName = ifelse(type == &quot;not significant&quot;,&quot;&quot;,targetName)) %&gt;%
  mutate(p.adj = ifelse(p.adj &lt;1e-5, 1e-5,p.adj))

#subset for mutation with at least one significant associations
plotMut &lt;- unique(filter(testRes, p.adj &lt;= 0.1)$mutName)

plotTab &lt;- plotTab %&gt;% dplyr::filter(mutName %in% plotMut) 
plotTab$type &lt;- factor(plotTab$type, levels = c(&quot;mutated&quot;, &quot;unmutated&quot;, &quot;not significant&quot;))

p &lt;- ggplot(data=plotTab, aes(x= mutName, y=-log10(p.adj),
                              col=type, label = varName))+ 
  geom_text_repel(position = pos, color = &quot;black&quot;, size= 6, force = 3) +
  geom_hline(yintercept = -log10(0.1), linetype=&quot;dotted&quot;, color = &quot;grey20&quot;) + 
  geom_point(size=3, position = pos) + 
  ylab(expression(-log[10]*&#39;(&#39;*adjusted~italic(&quot;P&quot;)~value*&#39;)&#39;)) + xlab(&quot;Mutation&quot;) +
  scale_color_manual(values = colList2) + 
  scale_y_continuous(trans = &quot;exp&quot;, limits = c(0,2.5), breaks = c(0,1,1.5,2)) + 
  theme_custom +
  #annotate(geom = &quot;text&quot;, x = 0.5, y = -log10(0.1) - 0.25, label = &quot;10% FDR&quot;, size=7, col = &quot;grey20&quot;) +
  coord_flip()  + labs(col = &quot;Higher dependence in&quot;)  +
  theme(legend.position = c(0.80,0.2),
        legend.background = element_rect(fill = NA),
        legend.text = element_text(size=14),
        legend.title = element_text(size=16),
        axis.title = element_text(size=18),
        axis.text = element_text(size=18))
plot(p)</code></pre>
<p><img src="figure/analysis_GDSC.Rmd/gdsc_pval_vis-1.png" width="768" style="display: block; margin: auto;" /></p>
<pre class="r"><code>#ggsave(&quot;test.pdf&quot;,height = 4, width = 8)</code></pre>
<p>Visualize significant associations using a heatmap</p>
<pre class="r"><code>plotTab &lt;- testRes %&gt;% dplyr::filter(mutName != &quot;cancer_type&quot;) %&gt;% 
  mutate(starSign = ifelse(p.adj &lt;=0.1, &quot;*&quot;, &quot;&quot;),
         pSign = -log10(p)*sign(FC))

#subset for mutation with at least one significant associations
plotTar &lt;- unique(filter(plotTab, p.adj &lt;= 0.1)$targetName)
plotMut &lt;- unique(filter(plotTab, p.adj &lt;= 0.1)$mutName)
plotTab &lt;- plotTab %&gt;% dplyr::filter( targetName %in% plotTar , mutName %in% plotMut) 


p &lt;- ggplot(data=plotTab, aes(y=mutName, x = targetName, fill=pSign)) +
  geom_tile(col = &quot;black&quot;) + geom_text(aes(label = starSign), size=5, vjust=0.5) +
  scale_fill_gradient2(low = &quot;#BC3C29FF&quot;, high = &quot;#0072B5FF&quot;, name = bquote(-log[10]*italic(&quot;P&quot;))) +
  theme_minimal() +
  theme(panel.grid.major = element_blank(),
        legend.text = element_text(size=14),
        legend.title = element_text(size=16),
        axis.title = element_text(size=18),
        axis.text = element_text(size=18)) +
  ylab(&quot;Mutations&quot;) + xlab(&quot;Proteins&quot;)

p</code></pre>
<p><img src="figure/analysis_GDSC.Rmd/gdsc_pval_Heatmap-1.png" width="768" style="display: block; margin: auto;" /></p>
<p>Visualization of exemplary association between NRAS mutation status and MAP2K2 dependence visualized in a beeswarm plot</p>
<pre class="r"><code>pList &lt;- plotDiffBox(testRes, result$coefMat, cell_anno_final, 0.05)
pList$MAP2K2_NRAS</code></pre>
<p><img src="figure/analysis_GDSC.Rmd/gdsc%20boxplot-1.png" width="480" style="display: block; margin: auto;" /></p>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span> Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 4.1.2 (2021-11-01)
Platform: x86_64-apple-darwin17.0 (64-bit)
Running under: macOS Big Sur 10.16

Matrix products: default
BLAS:   /Library/Frameworks/R.framework/Versions/4.1/Resources/lib/libRblas.0.dylib
LAPACK: /Library/Frameworks/R.framework/Versions/4.1/Resources/lib/libRlapack.dylib

locale:
[1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
 [1] forcats_0.5.1      stringr_1.4.0      dplyr_1.0.7        purrr_0.3.4       
 [5] readr_2.1.1        tidyr_1.1.4        tibble_3.1.6       tidyverse_1.3.1   
 [9] ggrepel_0.9.1      ggbeeswarm_0.6.0   ggplot2_3.3.5      pheatmap_1.0.12   
[13] RColorBrewer_1.1-2 depInfeR_0.1.0    

loaded via a namespace (and not attached):
 [1] httr_1.4.2         sass_0.4.0         jsonlite_1.7.2     splines_4.1.2     
 [5] foreach_1.5.1      modelr_0.1.8       bslib_0.3.1        assertthat_0.2.1  
 [9] highr_0.9          cellranger_1.1.0   vipor_0.4.5        yaml_2.2.1        
[13] pillar_1.6.4       backports_1.4.1    lattice_0.20-45    glue_1.5.1        
[17] rlist_0.4.6.2      digest_0.6.29      promises_1.2.0.1   rvest_1.0.2       
[21] colorspace_2.0-2   htmltools_0.5.2    httpuv_1.6.4       Matrix_1.4-0      
[25] pkgconfig_2.0.3    broom_0.7.10       haven_2.4.3        scales_1.1.1      
[29] later_1.3.0        tzdb_0.2.0         git2r_0.29.0       farver_2.1.0      
[33] generics_0.1.1     ellipsis_0.3.2     withr_2.4.3        cli_3.1.0         
[37] survival_3.2-13    magrittr_2.0.1     crayon_1.4.2       readxl_1.3.1      
[41] evaluate_0.14      fs_1.5.2           fansi_0.5.0        doParallel_1.0.16 
[45] xml2_1.3.3         beeswarm_0.4.0     tools_4.1.2        data.table_1.14.2 
[49] hms_1.1.1          lifecycle_1.0.1    matrixStats_0.61.0 reprex_2.0.1      
[53] munsell_0.5.0      glmnet_4.1-3       compiler_4.1.2     jquerylib_0.1.4   
[57] rlang_0.4.12       grid_4.1.2         rstudioapi_0.13    iterators_1.0.13  
[61] labeling_0.4.2     rmarkdown_2.11     gtable_0.3.0       codetools_0.2-18  
[65] DBI_1.1.1          R6_2.5.1           lubridate_1.8.0    knitr_1.36        
[69] fastmap_1.1.0      utf8_1.2.2         workflowr_1.7.0    rprojroot_2.0.2   
[73] shape_1.4.6        stringi_1.7.6      parallel_4.1.2     Rcpp_1.0.7        
[77] vctrs_0.3.8        dbplyr_2.1.1       tidyselect_1.1.1   xfun_0.29         </code></pre>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
https://docs.mathjax.org/en/latest/web/configuration.html. This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>




</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
