<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="Junyan Lu" />

<meta name="date" content="2021-10-26" />

<title>Analyzing CHEK1 signaling in CLL expression signatures</title>

<script src="site_libs/header-attrs-2.9/header-attrs.js"></script>
<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/master/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>








<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.tab('show');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">DepInfeR</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">



<h1 class="title toc-ignore">Analyzing CHEK1 signaling in CLL expression signatures</h1>
<h4 class="author">Junyan Lu</h4>
<h4 class="date">2021-10-26</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span> workflowr <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2021-11-05
</p>
<p>
<strong>Checks:</strong> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 6 <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> 1
</p>
<p>
<strong>Knit directory:</strong> <code>DepInfeR/analysis/</code> <span class="glyphicon glyphicon-question-sign" aria-hidden="true" title="This is the local directory in which the code in this file was executed."> </span>
</p>
<p>
This reproducible <a href="http://rmarkdown.rstudio.com">R Markdown</a> analysis was created with <a
  href="https://github.com/jdblischak/workflowr">workflowr</a> (version 1.6.2). The <em>Checks</em> tab describes the reproducibility checks that were applied when the results were created. The <em>Past versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguncommittedchanges"> <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> <strong>R Markdown file:</strong> uncommitted changes </a>
</p>
</div>
<div id="strongRMarkdownfilestronguncommittedchanges" class="panel-collapse collapse">
<div class="panel-body">
<p>The R Markdown is untracked by Git. To know which version of the R Markdown file created these results, you’ll want to first commit it to the Git repo. If you’re still working on the analysis, you can ignore this warning. When you’re finished, you can run <code>wflow_publish</code> to commit the R Markdown file and build the HTML.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the global environment can affect the analysis in your R Markdown file in unknown ways. For reproduciblity it’s best to always run the code in an empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20211005code"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Seed:</strong> <code>set.seed(20211005)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20211005code" class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20211005)</code> was run prior to running the code in the R Markdown file. Setting a seed ensures that any results that rely on randomness, e.g. subsampling or permutations, are reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Session information:</strong> recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be confident that you successfully produced the results during this run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrong43be8a7"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Repository version:</strong> 43be8a7 </a>
</p>
</div>
<div id="strongRepositoryversionstrong43be8a7" class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development and connecting the code version to the results is critical for reproducibility.
</p>
<p>
The results in this page were generated with repository version 43be8a7. See the <em>Past versions</em> tab to see a history of the changes made to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for the analysis have been committed to Git prior to generating the results (you can use <code>wflow_publish</code> or <code>wflow_git_commit</code>). workflowr only checks the R Markdown file, but you know if there are other scripts or data files that it depends on. Below is the status of the Git repository when the results were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .DS_Store
    Ignored:    .Rhistory
    Ignored:    .Rproj.user/
    Ignored:    analysis/.DS_Store
    Ignored:    analysis/.Rhistory
    Ignored:    data/.DS_Store
    Ignored:    output/.DS_Store

Untracked files:
    Untracked:  analysis/analysis_EMBL2016.Rmd
    Untracked:  analysis/analysis_GDSC.Rmd
    Untracked:  analysis/analysis_RNAseq.Rmd
    Untracked:  analysis/analysis_beatAML.Rmd
    Untracked:  analysis/process_EMBL2016.Rmd
    Untracked:  analysis/process_GDSC.Rmd
    Untracked:  analysis/process_beatAML.Rmd
    Untracked:  analysis/process_kinobeads.Rmd
    Untracked:  code/utils.R
    Untracked:  data/BeatAML/
    Untracked:  data/EMBL2016/
    Untracked:  data/GDSC/
    Untracked:  data/Kinobeads/
    Untracked:  data/RNAseq/
    Untracked:  manuscript/
    Untracked:  output/BeatAML_result.RData
    Untracked:  output/EMBL_result.RData
    Untracked:  output/EMBL_resultSub.RData
    Untracked:  output/GDSC_result.RData
    Untracked:  output/allTargets.rds
    Untracked:  output/inputs_BeatAML.RData
    Untracked:  output/inputs_EMBL.RData
    Untracked:  output/inputs_GDSC.RData

Unstaged changes:
    Modified:   README.md
    Modified:   _workflowr.yml
    Modified:   analysis/_site.yml
    Deleted:    analysis/about.Rmd
    Modified:   analysis/index.Rmd
    Deleted:    analysis/license.Rmd
    Deleted:    output/README.md

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not included in this status report because it is ok for generated content to have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">
<p>
There are no past versions. Publish this analysis with <code>wflow_publish()</code> to start tracking its development.
</p>
<hr>
</div>
</div>
</div>
<div id="load-packages" class="section level1">
<h1>Load packages</h1>
<p>Packages</p>
<pre class="r"><code>library(RColorBrewer)
library(DESeq2)
library(limma)
library(piano)
library(Biobase)
library(ggbeeswarm)
library(tidyverse)
source(&quot;../code/utils.R&quot;)
knitr::opts_chunk$set(dev=c(&quot;png&quot;,&quot;pdf&quot;))</code></pre>
</div>
<div id="bcr-triggering-signature-by-anti-igm-stimulation-for-390-min-gse3941" class="section level1">
<h1>BCR-triggering signature by anti-IgM stimulation for 390 min (GSE3941)</h1>
<pre class="r"><code>load(&quot;../data/RNAseq/gse39411.RData&quot;)
gmts = list(H = &quot;../data/RNAseq/h.all.v6.2.symbols.gmt&quot;)
setMap &lt;- read_tsv(&quot;../data/RNAseq/setToPathway.txt&quot;, col_types = &quot;cc&quot;)</code></pre>
<pre class="r"><code>#gse &lt;- gse39411[[1]]

#subset for only B-cell and CLL without transfection
#gse &lt;- gse[,gse$`transfected with:ch1` == &quot;none&quot; &amp;
#             gse$`time point (min):ch1` == &quot;T390&quot; &amp;
#             gse$`cell type:ch1` == &quot;chronic lymphocytic leukemia B-cell&quot;]
#save
#save(gse, file = &quot;../data/RNAseq/gse39411.RData&quot;)


#vst
gse.vst &lt;- gse
exprs(gse.vst) &lt;- limma::normalizeVSN(gse.vst)

patAnno &lt;- pData(gse.vst) %&gt;% rownames_to_column(&quot;sampleID&quot;) %&gt;%
  select(sampleID, description) %&gt;%
  separate(description, into = c(&quot;patID&quot;,&quot;stimulation&quot;,&quot;timePoint&quot;),sep = &quot;_&quot;) %&gt;%
  mutate(cellType = substr(patID,1,nchar(patID)-1)) %&gt;%
  mutate(cellType = ifelse(cellType == &quot;N&quot;, &quot;B-cell&quot;,cellType)) %&gt;%
  mutate(timePoint = abs(as.integer(gsub(&quot;T&quot;,&quot;&quot;,timePoint)))) %&gt;%
  mutate(stimulation = factor(stimulation, levels = c(&quot;US&quot;,&quot;S&quot;)))

pData(gse.vst) &lt;- patAnno[match(colnames(gse.vst), patAnno$sampleID),]</code></pre>
<p>Differential expression (CLL cells for 390 min)</p>
<pre class="r"><code>gse.test &lt;- gse.vst</code></pre>
<p>Enrichment</p>
<pre class="r"><code>highSet &lt;- c(&quot;UV response up&quot;, &quot;DNA repair&quot;, &quot;G2M checkpoint&quot;, &quot;E2F targets&quot;)

exprMat &lt;- exprs(gse.test)
designMat &lt;- model.matrix( ~ 1  + patID  + stimulation, pData(gse.test))

enrichRes &lt;- runCamera(exprMat, designMat, gmts$H, id = fData(gse.test)$`Gene Symbol`,
                       method = &quot;camera&quot;, pCut = 0.05, ifFDR = TRUE, removePrefix = &quot;HALLMARK_&quot;,
                       plotTitle = &quot;anti-IgM treatment&quot;,insideLegend = TRUE, setToHighlight = highSet, setMap = setMap)</code></pre>
<pre><code>Warning: Vectorized input to `element_text()` is not officially supported.
Results may be unexpected or may change in future versions of ggplot2.</code></pre>
<pre class="r"><code>plotEnrich.IgM &lt;- enrichRes$enrichPlot+ theme(plot.title = element_text(size=18, face = &quot;bold&quot;))
plotEnrich.IgM</code></pre>
<p><img src="figure/analysis_RNAseq.Rmd/RNAseq_IGM-1.png" width="768" style="display: block; margin: auto;" /></p>
</div>
<div id="pathway-enrichment-analysis-using-icgc-datasets" class="section level1">
<h1>Pathway enrichment analysis using ICGC datasets</h1>
<p>Load and pro-processing RNAseq dataset</p>
<pre class="r"><code>load(&quot;../data/RNAseq/ddsCampo.RData&quot;)
ddsCLL &lt;- ddsCampo[,!is.na(ddsCampo$IGHV) ]
ddsCLL &lt;- estimateSizeFactors(ddsCLL)
ddsCLL &lt;- ddsCLL[rowSums(counts(ddsCLL)) &gt; 10,]
ddsCLL.voom &lt;- ddsCLL
assay(ddsCLL.voom) &lt;- limma::voom(counts(ddsCLL), lib.size = ddsCLL$sizeFactor)$E</code></pre>
<div id="enrichment-analysis" class="section level3">
<h3>Enrichment analysis</h3>
<pre class="r"><code>ighv &lt;- factor(ddsCLL.voom$IGHV, levels = c(&quot;mutated&quot;,&quot;unmutated&quot;))
designMat &lt;- model.matrix(~ 1 + ighv)
enrichRes &lt;- runCamera(assay(ddsCLL.voom), designMat, gmts$H, 
                       id = rowData(ddsCLL.voom)$symbol, 
                       method = &quot;camera&quot;, pCut = 0.01, ifFDR = TRUE, removePrefix = &quot;HALLMARK_&quot;,
                       plotTitle = &quot;ICGC-CLL cohort (Hallmarks, 5% FDR)&quot;, 
                       setToHighlight = highSet, setMap = setMap, insideLegend = TRUE)</code></pre>
<pre><code>Warning: Vectorized input to `element_text()` is not officially supported.
Results may be unexpected or may change in future versions of ggplot2.</code></pre>
<pre class="r"><code>enrichICGC &lt;- enrichRes$enrichPlot
enrichICGC</code></pre>
<p><img src="figure/analysis_RNAseq.Rmd/RNAseq_ICGC-1.png" width="768" style="display: block; margin: auto;" /></p>
</div>
<div id="differentially-expressed-genes" class="section level3">
<h3>Differentially expressed genes</h3>
<pre class="r"><code>ddsCLL$IGHV &lt;- factor(ddsCLL$IGHV, levels = c(&quot;unmutated&quot;,&quot;mutated&quot;))
design(ddsCLL) &lt;- ~ IGHV
deRes &lt;- DESeq(ddsCLL)</code></pre>
<pre><code>using pre-existing size factors</code></pre>
<pre><code>estimating dispersions</code></pre>
<pre><code>gene-wise dispersion estimates</code></pre>
<pre><code>mean-dispersion relationship</code></pre>
<pre><code>final dispersion estimates</code></pre>
<pre><code>fitting model and testing</code></pre>
<pre><code>-- replacing outliers and refitting for 2354 genes
-- DESeq argument &#39;minReplicatesForReplace&#39; = 7 
-- original counts are preserved in counts(dds)</code></pre>
<pre><code>estimating dispersions</code></pre>
<pre><code>fitting model and testing</code></pre>
<pre class="r"><code>resTab &lt;- results(deRes, tidy = TRUE) %&gt;%
  mutate(symbol = rowData(ddsCLL[row,])$symbol)</code></pre>
<div class="alert alert-warning">
<p>
<strong>Warning:</strong> The above code chunk cached its results, but it won’t be re-run if previous chunks it depends on are updated. If you need to use caching, it is highly recommended to also set <code>knitr::opts_chunk$set(autodep = TRUE)</code> at the top of the file (in a chunk that is not cached). Alternatively, you can customize the option <code>dependson</code> for each individual chunk that is cached. Using either <code>autodep</code> or <code>dependson</code> will remove this warning. See the <a href="https://yihui.name/knitr/options/#cache"
>knitr cache options</a> for more details.
</p>
</div>
<pre class="r"><code>geneList &lt;- c(&quot;CDC25A&quot;,&quot;CDK1&quot;,&quot;CCNE1&quot;)
sigTab &lt;- filter(resTab, symbol %in% geneList)
pList &lt;- lapply(seq(nrow(sigTab)), function(i) {
    
    symbol &lt;- sigTab[i,]$symbol
    pval &lt;- sigTab[i,]$pvalue
    id &lt;- sigTab[i,]$row
    
    plotTab &lt;- tibble(expr = log2(counts(ddsCLL)[id,]+1),
                      IGHV.status = str_to_title(ddsCLL$IGHV),
                      patID = colnames(ddsCLL))
      
    numTab &lt;- group_by(plotTab, IGHV.status) %&gt;%
      summarise(n=length(patID))

    plotTab &lt;- left_join(plotTab, numTab, by = &quot;IGHV.status&quot;) %&gt;%
      mutate(mutNum = sprintf(&quot;%s\n(n=%s)&quot;, IGHV.status, n)) %&gt;%
      mutate(mutNum = factor(mutNum, levels = unique(mutNum)))

    titleText &lt;- sprintf(&quot;%s&quot;, symbol)
    pval &lt;- formatNum(pval, digits = 1, format=&quot;e&quot;)
    titleText &lt;- bquote(atop(.(titleText), &quot;(&quot;~italic(&quot;P&quot;)~&quot;=&quot;~.(pval)~&quot;)&quot;))

    ggplot(plotTab, aes(x = mutNum,y = expr)) +
      stat_boxplot(geom = &quot;errorbar&quot;, width = 0.3) +
      geom_boxplot(outlier.shape = NA, col=&quot;black&quot;, width=0.4) +
      geom_beeswarm(cex=2, size =1, aes(col = mutNum)) + 
      theme_classic() +
      xlab(&quot;&quot;) + ylab(&quot;log2(RNAseq counts)&quot;) + ggtitle(titleText) + xlab(&quot;&quot;) +
      scale_color_manual(values = c(&quot;#0072B5FF&quot;,&quot;#BC3C29FF&quot;)) +
      theme(axis.line.x = element_blank(), axis.ticks.x = element_blank(),
            axis.title = element_text(size=18),
            axis.text = element_text(size=18),
            plot.title = element_text(size= 18, face = &quot;bold&quot;, hjust = 0.5),
            legend.position = &quot;none&quot;,
            axis.title.x = element_text(face=&quot;bold&quot;))
})
noY &lt;- theme(axis.title.y = element_blank())
cowplot::plot_grid(pList[[1]],pList[[2]]+noY, pList[[3]] + noY, nrow=1,
                   rel_widths = c(1.1,1,1))</code></pre>
<p><img src="figure/analysis_RNAseq.Rmd/CHK_expr-1.png" width="960" style="display: block; margin: auto;" /></p>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span> Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 4.0.2 (2020-06-22)
Platform: x86_64-apple-darwin17.0 (64-bit)
Running under: macOS  10.16

Matrix products: default
BLAS:   /Library/Frameworks/R.framework/Versions/4.0/Resources/lib/libRblas.dylib
LAPACK: /Library/Frameworks/R.framework/Versions/4.0/Resources/lib/libRlapack.dylib

locale:
[1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8

attached base packages:
[1] parallel  stats4    stats     graphics  grDevices utils     datasets 
[8] methods   base     

other attached packages:
 [1] forcats_0.5.1               stringr_1.4.0              
 [3] dplyr_1.0.7                 purrr_0.3.4                
 [5] readr_2.0.0                 tidyr_1.1.3                
 [7] tibble_3.1.3                tidyverse_1.3.1            
 [9] ggbeeswarm_0.6.0            ggplot2_3.3.5              
[11] piano_2.4.0                 limma_3.44.3               
[13] DESeq2_1.28.1               SummarizedExperiment_1.18.2
[15] DelayedArray_0.14.1         matrixStats_0.60.0         
[17] Biobase_2.48.0              GenomicRanges_1.40.0       
[19] GenomeInfoDb_1.24.2         IRanges_2.22.2             
[21] S4Vectors_0.26.1            BiocGenerics_0.34.0        
[23] RColorBrewer_1.1-2         

loaded via a namespace (and not attached):
  [1] readxl_1.3.1           backports_1.2.1        fastmatch_1.1-3       
  [4] workflowr_1.6.2        igraph_1.2.6           shinydashboard_0.7.1  
  [7] splines_4.0.2          BiocParallel_1.22.0    digest_0.6.27         
 [10] htmltools_0.5.1.1      fansi_0.5.0            magrittr_2.0.1        
 [13] memoise_2.0.0          cluster_2.1.2          tzdb_0.1.2            
 [16] annotate_1.66.0        modelr_0.1.8           vroom_1.5.3           
 [19] colorspace_2.0-2       blob_1.2.2             rvest_1.0.1           
 [22] haven_2.4.1            xfun_0.24              crayon_1.4.1          
 [25] RCurl_1.98-1.3         jsonlite_1.7.2         genefilter_1.70.0     
 [28] survival_3.2-11        glue_1.4.2             gtable_0.3.0          
 [31] zlibbioc_1.34.0        XVector_0.28.0         scales_1.1.1          
 [34] vsn_3.56.0             DBI_1.1.1              relations_0.6-9       
 [37] Rcpp_1.0.7             xtable_1.8-4           bit_4.0.4             
 [40] preprocessCore_1.50.0  DT_0.18                htmlwidgets_1.5.3     
 [43] httr_1.4.2             fgsea_1.14.0           gplots_3.1.1          
 [46] ellipsis_0.3.2         farver_2.1.0           pkgconfig_2.0.3       
 [49] XML_3.99-0.6           sass_0.4.0             dbplyr_2.1.1          
 [52] locfit_1.5-9.4         utf8_1.2.2             tidyselect_1.1.1      
 [55] labeling_0.4.2         rlang_0.4.11           later_1.2.0           
 [58] AnnotationDbi_1.50.3   munsell_0.5.0          cellranger_1.1.0      
 [61] tools_4.0.2            visNetwork_2.0.9       cachem_1.0.5          
 [64] cli_3.0.1              generics_0.1.0         RSQLite_2.2.7         
 [67] broom_0.7.9            evaluate_0.14          fastmap_1.1.0         
 [70] yaml_2.2.1             knitr_1.33             bit64_4.0.5           
 [73] fs_1.5.0               caTools_1.18.2         mime_0.11             
 [76] slam_0.1-48            xml2_1.3.2             compiler_4.0.2        
 [79] rstudioapi_0.13        beeswarm_0.4.0         affyio_1.58.0         
 [82] marray_1.66.0          reprex_2.0.0           geneplotter_1.66.0    
 [85] bslib_0.2.5.1          stringi_1.7.3          highr_0.9             
 [88] lattice_0.20-44        Matrix_1.3-4           shinyjs_2.0.0         
 [91] vctrs_0.3.8            pillar_1.6.2           lifecycle_1.0.0       
 [94] BiocManager_1.30.16    jquerylib_0.1.4        cowplot_1.1.1         
 [97] data.table_1.14.0      bitops_1.0-7           httpuv_1.6.1          
[100] R6_2.5.0               affy_1.66.0            promises_1.2.0.1      
[103] KernSmooth_2.23-20     gridExtra_2.3          vipor_0.4.5           
[106] codetools_0.2-18       gtools_3.9.2           assertthat_0.2.1      
[109] rprojroot_2.0.2        withr_2.4.2            GenomeInfoDbData_1.2.3
[112] hms_1.1.0              grid_4.0.2             rmarkdown_2.9         
[115] git2r_0.28.0           sets_1.0-18            shiny_1.6.0           
[118] lubridate_1.7.10      </code></pre>
</div>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
http://docs.mathjax.org/en/latest/configuration.html.  This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>




</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
