<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="Junyan Lu" />

<meta name="date" content="2021-10-05" />

<title>Pre-processing GDSC drug screen datasets</title>

<script src="site_libs/header-attrs-2.11/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/main/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>








<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.tab('show');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">DepInfeR</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">



<h1 class="title toc-ignore">Pre-processing GDSC drug screen datasets</h1>
<h4 class="author">Junyan Lu</h4>
<h4 class="date">2021-10-05</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span> workflowr <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2021-12-24
</p>
<p>
<strong>Checks:</strong> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 6 <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> 1
</p>
<p>
<strong>Knit directory:</strong> <code>DepInfeR/analysis/</code> <span class="glyphicon glyphicon-question-sign" aria-hidden="true" title="This is the local directory in which the code in this file was executed."> </span>
</p>
<p>
This reproducible <a href="https://rmarkdown.rstudio.com">R Markdown</a> analysis was created with <a
  href="https://github.com/workflowr/workflowr">workflowr</a> (version 1.7.0). The <em>Checks</em> tab describes the reproducibility checks that were applied when the results were created. The <em>Past versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguncommittedchanges"> <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> <strong>R Markdown file:</strong> uncommitted changes </a>
</p>
</div>
<div id="strongRMarkdownfilestronguncommittedchanges" class="panel-collapse collapse">
<div class="panel-body">
<p>The R Markdown is untracked by Git. To know which version of the R Markdown file created these results, you’ll want to first commit it to the Git repo. If you’re still working on the analysis, you can ignore this warning. When you’re finished, you can run <code>wflow_publish</code> to commit the R Markdown file and build the HTML.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the global environment can affect the analysis in your R Markdown file in unknown ways. For reproduciblity it’s best to always run the code in an empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20211005code"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Seed:</strong> <code>set.seed(20211005)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20211005code" class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20211005)</code> was run prior to running the code in the R Markdown file. Setting a seed ensures that any results that rely on randomness, e.g. subsampling or permutations, are reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Session information:</strong> recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be confident that you successfully produced the results during this run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrong43be8a7"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Repository version:</strong> 43be8a7 </a>
</p>
</div>
<div id="strongRepositoryversionstrong43be8a7" class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development and connecting the code version to the results is critical for reproducibility.
</p>
<p>
The results in this page were generated with repository version 43be8a7. See the <em>Past versions</em> tab to see a history of the changes made to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for the analysis have been committed to Git prior to generating the results (you can use <code>wflow_publish</code> or <code>wflow_git_commit</code>). workflowr only checks the R Markdown file, but you know if there are other scripts or data files that it depends on. Below is the status of the Git repository when the results were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .DS_Store
    Ignored:    .Rhistory
    Ignored:    .Rproj.user/
    Ignored:    analysis/.DS_Store
    Ignored:    analysis/.Rhistory
    Ignored:    analysis/analysis_RNAseq_cache/
    Ignored:    analysis/figure/
    Ignored:    data/.DS_Store
    Ignored:    output/.DS_Store

Untracked files:
    Untracked:  analysis/analysis_EMBL2016.Rmd
    Untracked:  analysis/analysis_GDSC.Rmd
    Untracked:  analysis/analysis_RNAseq.Rmd
    Untracked:  analysis/analysis_beatAML.Rmd
    Untracked:  analysis/process_EMBL2016.Rmd
    Untracked:  analysis/process_GDSC.Rmd
    Untracked:  analysis/process_beatAML.Rmd
    Untracked:  analysis/process_kinobeads.Rmd
    Untracked:  code/utils.R
    Untracked:  data/BeatAML/
    Untracked:  data/EMBL2016/
    Untracked:  data/GDSC/
    Untracked:  data/Kinobeads/
    Untracked:  data/RNAseq/
    Untracked:  manuscript/
    Untracked:  output/BeatAML_result.RData
    Untracked:  output/EMBL_result.RData
    Untracked:  output/EMBL_resultSub.RData
    Untracked:  output/GDSC_result.RData
    Untracked:  output/allTargets.rds
    Untracked:  output/inputs_BeatAML.RData
    Untracked:  output/inputs_EMBL.RData
    Untracked:  output/inputs_GDSC.RData

Unstaged changes:
    Modified:   README.md
    Modified:   _workflowr.yml
    Modified:   analysis/_site.yml
    Deleted:    analysis/about.Rmd
    Modified:   analysis/index.Rmd
    Deleted:    analysis/license.Rmd
    Deleted:    output/README.md

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not included in this status report because it is ok for generated content to have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">
<p>
There are no past versions. Publish this analysis with <code>wflow_publish()</code> to start tracking its development.
</p>
<hr>
</div>
</div>
</div>
<div id="introduction" class="section level1">
<h1>Introduction</h1>
<p>This document shows the pre-processing steps of the GDSC cancer cell line screening dataset from <a href="https://www.cancerrxgene.org/" class="uri">https://www.cancerrxgene.org/</a>. A subset of leukemia and breast cancer cell lines was chosen for this analysis (called set1 hereinafter). The analyzed cancer types were</p>
<ul>
<li><p>Diffuse Large B-Cell Lymphoma (DLBC)</p></li>
<li><p>Acute lymphocytic leukemia (ALL)</p></li>
<li><p>Acute myeloid leukemia (AML)</p></li>
<li><p>Breast carcinoma (BRCAHer+ / BRCAHer-)</p></li>
</ul>
<p>The Her2 status was annotated manually.</p>
<hr />
</div>
<div id="load-packages" class="section level1">
<h1>Load packages</h1>
<p>Packages</p>
<pre class="r"><code>library(depInfeR)
library(missForest)
library(tidyverse)
source(&quot;../code/utils.R&quot;)
knitr::opts_chunk$set(dev = c(&quot;png&quot;,&quot;pdf&quot;))</code></pre>
</div>
<div id="read-data-sets" class="section level1">
<h1>Read data sets</h1>
<p>Load pre-processed kinobead table table</p>
<pre class="r"><code>tarList &lt;- readRDS(&quot;../output/allTargets.rds&quot;)</code></pre>
<p>Load GDSC raw data tables</p>
<pre class="r"><code># GDSC cell line screening data (for each of the cancer types and the table with the drug synonyms)
cancerxgene_ALL &lt;- read_csv(&quot;../data/GDSC/cancerxgene_ALL_IC_GDSC1.csv&quot;)
cancerxgene_DLBC &lt;- read_csv(&quot;../data/GDSC/cancerxgene_DLBC_IC_GDSC1.csv&quot;)
cancerxgene_AML &lt;- read_csv(&quot;../data/GDSC/cancerxgene_AML_IC_GDSC1.csv&quot;)
cancerxgene_BRCA &lt;- read_csv(&quot;../data/GDSC/cancerxgene_BRCA_IC_GDSC1.csv&quot;)
cancerxgene_syn &lt;- read_csv(&quot;../data/GDSC/GDSC1_druglist_pubchem.csv&quot;)


# cancer cell line genetic background annotation
# mutations
BRCA_mut &lt;- read_csv(&quot;../data/GDSC/BRCA_genetic.csv&quot;, col_names = TRUE, col_types = cols(.default = col_factor()))
ALL_mut &lt;- read_csv(&quot;../data/GDSC/ALL_genetics.csv&quot;, col_names = TRUE, col_types = cols(.default = col_factor()))
LAML_mut &lt;- read_csv(&quot;../data/GDSC/LAML_genetic.csv&quot;, col_names = TRUE, col_types = cols(.default = col_factor()))
DLBC_mut &lt;- read_csv(&quot;../data/GDSC/DLBC_genetic.csv&quot;, col_names = TRUE, col_types = cols(.default = col_factor()))</code></pre>
</div>
<div id="preprocess-drug-target-dataset" class="section level1">
<h1>Preprocess drug target dataset</h1>
<p>In order to apply the regression algorithm both datasets need to be filtered for matching drugs. Therefore, the used drug names need to be aligned and the datatables need to be filtered for the overlapping drug names. To find as many matching drug names as possible, we firstly add the synonym columns that we found in different tables for each of the datasets to the datatables. Secondly, we calculate the Hamming Distance between the drug names (including the synonyms columns) to find matching drug names with only slightly different spelling.</p>
<div id="preprocess-cancer-cell-line-drug-screen" class="section level2">
<h2>Preprocess cancer cell line drug screen</h2>
<p>Combine drug-cell line matrices of GDSC cancer cell line set (set1)</p>
<pre class="r"><code>bind_rows_keep_factors &lt;- function(...) {
  ## Identify all factors
  factors &lt;- unique(unlist(
    map(list(...), ~ select_if(..., is.factor) %&gt;% names())
  ))
  ## Bind dataframes, convert characters back to factors
  suppressWarnings(bind_rows(...)) %&gt;% 
    mutate_at(dplyr::vars(one_of(factors)), factor)  
}

cancerxgene_set1 &lt;- bind_rows_keep_factors(cancerxgene_ALL, cancerxgene_AML, cancerxgene_DLBC, cancerxgene_BRCA)</code></pre>
<p>Attach synonyms to cancerxgene table</p>
<pre class="r"><code>cancerxgene_set1$synonyms &lt;- cancerxgene_syn$Synonyms[match(cancerxgene_set1$`Drug name`, cancerxgene_syn$Name)]</code></pre>
<p>Process drug names of cancerxgene table</p>
<pre class="r"><code>cancerxgene_set1 &lt;- mutate(cancerxgene_set1, `Drug name` = tolower(`Drug name`)) %&gt;%
  mutate(`Drug name` = gsub(&quot;[- ]&quot;,&quot;&quot;,`Drug name`))  %&gt;% mutate(`Drug name`= gsub(&quot; *\\(.*?\\) *&quot;, &quot;&quot;,`Drug name`)) </code></pre>
</div>
<div id="find-overlapping-drugs-between-datasets" class="section level2">
<h2>Find overlapping drugs between datasets</h2>
<p>Find overlapped drugs by drug names</p>
<pre class="r"><code>overDrug_cancerx_set1_name &lt;- intersect(tarList$Drug, cancerxgene_set1$`Drug name`)</code></pre>
<p>Assign manually identified synonyms</p>
<pre class="r"><code>tarList &lt;- mutate(tarList, Drug = ifelse(Drug==&quot;alvocidib&quot;, &quot;flavopiridol&quot;, Drug))
tarList &lt;- mutate(tarList, Drug = ifelse(Drug==&quot;canertinib&quot;, &quot;ci1033&quot;, Drug))
tarList &lt;- mutate(tarList, Drug = ifelse(Drug==&quot;dacomitinib&quot;, &quot;pf00299804&quot;, Drug))
tarList &lt;- mutate(tarList, Drug = ifelse(Drug==&quot;nintedanib&quot;, &quot;bibf1120&quot;, Drug))</code></pre>
<p>Get the final overlapped drug list</p>
<pre class="r"><code>finalList &lt;- intersect(tarList$Drug, cancerxgene_set1$`Drug name`)</code></pre>
<p>Rename drug column in cancerxgene</p>
<pre class="r"><code>cancerxgene_set1 &lt;- dplyr::rename(cancerxgene_set1, Drug = `Drug name`)
cancerxgene_set1_druglist &lt;- filter(cancerxgene_set1, !is.na(`Drug Id`), !duplicated(`Drug Id`), !duplicated(Drug))</code></pre>
</div>
<div id="match-drug-ids-and-create-drug-target-affinity-matrix" class="section level2">
<h2>Match drug IDs and create drug-target affinity matrix</h2>
<p>Combine the lists</p>
<pre class="r"><code>targets &lt;- left_join(tarList, cancerxgene_set1_druglist, by = &quot;Drug&quot;) %&gt;% dplyr::select(Drug, `Drug Id`, `Target Classification`, EC50,`Apparent Kd`, `Gene Name`) %&gt;%
  filter(!is.na(Drug)) %&gt;%
  filter(Drug %in% finalList) </code></pre>
<p>How many drugs?</p>
<pre class="r"><code>length(unique(targets$Drug))</code></pre>
<pre><code>[1] 68</code></pre>
<p>Change column names</p>
<pre class="r"><code>colnames(targets) &lt;- c(&quot;drugName&quot;, &quot;drugID&quot;, &quot;targetClassification&quot;,&quot;EC50&quot;,&quot;Kd&quot;,&quot;targetName&quot;)</code></pre>
<p>Turn target table into drug-target affinity matrix</p>
<pre class="r"><code>tarMat_kd &lt;- dplyr::filter(targets, targetClassification == &quot;High confidence&quot;) %&gt;% 
    dplyr::select(drugName, targetName, Kd) %&gt;% 
    spread(key = &quot;targetName&quot;, value = &quot;Kd&quot;) %&gt;%
    remove_rownames() %&gt;% column_to_rownames(&quot;drugName&quot;) %&gt;% as.matrix()</code></pre>
<p>Function to transform Kd values (using arctan function)</p>
<pre class="r"><code>testTab &lt;- tibble(x = seq(-6,2,length.out = 20)) %&gt;% mutate(y = arcTrans(x, b=2, g=3))
ggplot(testTab, aes(x=x,y=y)) + geom_point() + 
  xlab(bquote(&quot;original &quot;~-log[10]*&quot;(Kd) value&quot;)) + ylab(&quot;transformed value&quot;) +
  theme_custom</code></pre>
<p><img src="figure/process_GDSC.Rmd/supp_arcTrans-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>As a pre-processing of the drug-protein affinity matrix with kd values (or optionally other affinity measurement values at roughly normal distribution) we chose to perform the following steps:</p>
<ul>
<li>log-transform kd values (KdAsInput = TRUE)</li>
<li>arctan-transform log(kd) values (KdAsInput = TRUE)</li>
<li>check target similarity and remove highly correlated targets (removeCorrelated = TRUE)</li>
</ul>
<pre class="r"><code>ProcessTargetResults &lt;- processTarget(tarMat_kd, KdAsInput  = TRUE , removeCorrelated  = TRUE)</code></pre>
</div>
<div id="visualize-target-groups-of-highly-correlated-targets-in-network-map" class="section level2">
<h2>Visualize target groups of highly correlated targets in network map</h2>
<pre class="r"><code>load(&quot;../output/GDSC_result.RData&quot;)
CancerxTargets&lt;- rowSums(result$freqMat)
CancerxTargets &lt;- names(CancerxTargets[CancerxTargets&gt;0])

#plot network
plotTarGroups(ProcessTargetResults, CancerxTargets)</code></pre>
<p><img src="figure/process_GDSC.Rmd/targetNet_GDSC-1.png" width="768" style="display: block; margin: auto;" /></p>
</div>
</div>
<div id="prepare-drug-response-matrix-drug-x-cell-line" class="section level1">
<h1>Prepare drug response matrix (drug X cell line)</h1>
<div id="prepare-response-matrix-using-the-z-score" class="section level2">
<h2>Prepare response matrix using the z-score</h2>
<p>The z-score was chosen as a suitable measurement value for our drug screening response matrix as it corresponds to a normalization for each drug over all cell lines. When working with AUC or IC50 values, a suitable normalization of the values is recommended. In this analysis I used the z-score of the AUC values.</p>
<pre class="r"><code>sanger_viab &lt;- dplyr::filter(cancerxgene_set1, `Drug Id` %in%targetsGDSC$drugID) %&gt;%
  dplyr::select(Drug, `Drug Id`, `Cell line name` , `Cosmic sample Id`, `TCGA classification`, IC50, AUC, `Max conc`,RMSE, `Z score`)
sanger_matrix &lt;- sanger_viab %&gt;% dplyr::select(Drug, `Cell line name`, AUC) %&gt;% 
  tidyr::spread(key = `Cell line name`, value = AUC) %&gt;%
  remove_rownames() %&gt;% column_to_rownames(&quot;Drug&quot;) %&gt;%
  as.matrix()</code></pre>
</div>
<div id="assessment-of-missing-values" class="section level2">
<h2>Assessment of missing values</h2>
<p>As we have some missing values in our response matrix, we check the distribution of our missing values across all cell lines</p>
<pre class="r"><code>missTab &lt;- data.frame(NA_cutoff = character(0), remain_celllines = character(0), stringsAsFactors = FALSE)
      
for (i in 0 : 138) {
  a &lt;- dim(sanger_matrix[,colSums(is.na(sanger_matrix)) &lt;= i])[2]
  missTab [i,] &lt;- c(i, a)
}
#missTab
#plot(missTab, type = &quot;l&quot;)</code></pre>
<p>From looking at the missing value distribution, we choose cell lines with a maximum of 24 missing values per cell line (= 35%) as usable for the MissForest imputation method.</p>
</div>
<div id="subset-for-cell-lines-with-less-than-24-missing-values-based-on-assessment-above" class="section level2">
<h2>Subset for cell lines with less than 24 missing values (based on assessment above)</h2>
<pre class="r"><code>sanger_mat_subset &lt;- sanger_matrix[,colSums(is.na(sanger_matrix)) &lt;= 24]</code></pre>
</div>
<div id="missforest-imputation" class="section level2">
<h2>MissForest imputation</h2>
<pre class="r"><code>impRes &lt;- missForest(t(sanger_mat_subset))</code></pre>
<pre><code>  missForest iteration 1 in progress...done!
  missForest iteration 2 in progress...done!
  missForest iteration 3 in progress...done!
  missForest iteration 4 in progress...done!
  missForest iteration 5 in progress...done!
  missForest iteration 6 in progress...done!
  missForest iteration 7 in progress...done!</code></pre>
<pre class="r"><code>imp_missforest &lt;- impRes$ximp

sanger_mat_forest &lt;- t(imp_missforest)
colnames(sanger_mat_forest) &lt;- colnames(sanger_mat_forest)
rownames(sanger_mat_forest) &lt;- rownames(sanger_mat_forest)</code></pre>
</div>
<div id="column-wise-z-score" class="section level2">
<h2>Column-wise z-score</h2>
<pre class="r"><code>sanger_mat_forest.scale &lt;- t(mscale(t(sanger_mat_forest)))</code></pre>
</div>
</div>
<div id="read-cell-line-annotations" class="section level1">
<h1>Read cell line annotations</h1>
<pre class="r"><code>mutation_GDSC &lt;- readxl::read_xlsx(&quot;../data/GDSC/mutation_GDSC.xlsx&quot;) %&gt;%
  mutate(TCGA.classification = str_replace(TCGA.classification, &quot;LAML&quot;,&quot;AML&quot;)) %&gt;%
  data.frame() %&gt;%
  column_to_rownames(&quot;cellLine&quot;) </code></pre>
</div>
<div id="save-pre-processed-dataset" class="section level1">
<h1>Save pre-processed dataset</h1>
<pre class="r"><code>ProcessTargetResults_GDSC &lt;- ProcessTargetResults
tarMat_GDSC &lt;- ProcessTargetResults$targetMatrix
viabMat_GDSC &lt;- sanger_mat_forest.scale[rownames(tarMat_GDSC),]
save(tarMat_GDSC, viabMat_GDSC, ProcessTargetResults_GDSC, mutation_GDSC,
     file = &quot;../output/inputs_GDSC.RData&quot;)</code></pre>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span> Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 4.1.2 (2021-11-01)
Platform: x86_64-apple-darwin17.0 (64-bit)
Running under: macOS Big Sur 10.16

Matrix products: default
BLAS:   /Library/Frameworks/R.framework/Versions/4.1/Resources/lib/libRblas.0.dylib
LAPACK: /Library/Frameworks/R.framework/Versions/4.1/Resources/lib/libRlapack.dylib

locale:
[1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
 [1] forcats_0.5.1       stringr_1.4.0       dplyr_1.0.7        
 [4] purrr_0.3.4         readr_2.1.1         tidyr_1.1.4        
 [7] tibble_3.1.6        ggplot2_3.3.5       tidyverse_1.3.1    
[10] missForest_1.4      itertools_0.1-3     iterators_1.0.13   
[13] foreach_1.5.1       randomForest_4.6-14 depInfeR_0.1.0     

loaded via a namespace (and not attached):
  [1] colorspace_2.0-2    ellipsis_0.3.2      rprojroot_2.0.2    
  [4] htmlTable_2.3.0     corpcor_1.6.10      base64enc_0.1-3    
  [7] fs_1.5.2            rstudioapi_0.13     lavaan_0.6-9       
 [10] farver_2.1.0        bit64_4.0.5         fansi_0.5.0        
 [13] lubridate_1.8.0     xml2_1.3.3          codetools_0.2-18   
 [16] splines_4.1.2       mnormt_2.0.2        doParallel_1.0.16  
 [19] knitr_1.36          glasso_1.11         rlist_0.4.6.2      
 [22] Formula_1.2-4       jsonlite_1.7.2      workflowr_1.7.0    
 [25] broom_0.7.10        cluster_2.1.2       dbplyr_2.1.1       
 [28] png_0.1-7           compiler_4.1.2      httr_1.4.2         
 [31] backports_1.4.1     assertthat_0.2.1    Matrix_1.4-0       
 [34] fastmap_1.1.0       cli_3.1.0           later_1.3.0        
 [37] htmltools_0.5.2     tools_4.1.2         igraph_1.2.10      
 [40] gtable_0.3.0        glue_1.5.1          reshape2_1.4.4     
 [43] Rcpp_1.0.7          cellranger_1.1.0    jquerylib_0.1.4    
 [46] vctrs_0.3.8         nlme_3.1-153        psych_2.1.9        
 [49] xfun_0.29           rvest_1.0.2         lifecycle_1.0.1    
 [52] gtools_3.9.2        scales_1.1.1        vroom_1.5.7        
 [55] hms_1.1.1           promises_1.2.0.1    parallel_4.1.2     
 [58] RColorBrewer_1.1-2  yaml_2.2.1          pbapply_1.5-0      
 [61] gridExtra_2.3       sass_0.4.0          rpart_4.1-15       
 [64] latticeExtra_0.6-29 stringi_1.7.6       highr_0.9          
 [67] checkmate_2.0.0     shape_1.4.6         rlang_0.4.12       
 [70] pkgconfig_2.0.3     matrixStats_0.61.0  evaluate_0.14      
 [73] lattice_0.20-45     htmlwidgets_1.5.4   labeling_0.4.2     
 [76] bit_4.0.4           tidyselect_1.1.1    ggsci_2.9          
 [79] plyr_1.8.6          magrittr_2.0.1      R6_2.5.1           
 [82] generics_0.1.1      Hmisc_4.6-0         DBI_1.1.1          
 [85] pillar_1.6.4        haven_2.4.3         foreign_0.8-81     
 [88] withr_2.4.3         abind_1.4-5         survival_3.2-13    
 [91] nnet_7.3-16         modelr_0.1.8        crayon_1.4.2       
 [94] fdrtool_1.2.17      utf8_1.2.2          tmvnsim_1.0-2      
 [97] tzdb_0.2.0          rmarkdown_2.11      jpeg_0.1-9         
[100] grid_4.1.2          readxl_1.3.1        qgraph_1.9         
[103] pbivnorm_0.6.0      data.table_1.14.2   git2r_0.29.0       
[106] reprex_2.0.1        digest_0.6.29       httpuv_1.6.4       
[109] stats4_4.1.2        munsell_0.5.0       glmnet_4.1-3       
[112] bslib_0.3.1        </code></pre>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
https://docs.mathjax.org/en/latest/web/configuration.html. This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>




</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
