<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="Junyan Lu" />

<meta name="date" content="2021-10-05" />

<title>Pre-processing EMBL2016 drug screen datasets</title>

<script src="site_libs/header-attrs-2.9/header-attrs.js"></script>
<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/master/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>








<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.tab('show');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">DepInfeR</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">



<h1 class="title toc-ignore">Pre-processing EMBL2016 drug screen datasets</h1>
<h4 class="author">Junyan Lu</h4>
<h4 class="date">2021-10-05</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span> workflowr <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2021-11-05
</p>
<p>
<strong>Checks:</strong> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 6 <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> 1
</p>
<p>
<strong>Knit directory:</strong> <code>DepInfeR/analysis/</code> <span class="glyphicon glyphicon-question-sign" aria-hidden="true" title="This is the local directory in which the code in this file was executed."> </span>
</p>
<p>
This reproducible <a href="http://rmarkdown.rstudio.com">R Markdown</a> analysis was created with <a
  href="https://github.com/jdblischak/workflowr">workflowr</a> (version 1.6.2). The <em>Checks</em> tab describes the reproducibility checks that were applied when the results were created. The <em>Past versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguncommittedchanges"> <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> <strong>R Markdown file:</strong> uncommitted changes </a>
</p>
</div>
<div id="strongRMarkdownfilestronguncommittedchanges" class="panel-collapse collapse">
<div class="panel-body">
<p>The R Markdown is untracked by Git. To know which version of the R Markdown file created these results, you’ll want to first commit it to the Git repo. If you’re still working on the analysis, you can ignore this warning. When you’re finished, you can run <code>wflow_publish</code> to commit the R Markdown file and build the HTML.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the global environment can affect the analysis in your R Markdown file in unknown ways. For reproduciblity it’s best to always run the code in an empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20211005code"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Seed:</strong> <code>set.seed(20211005)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20211005code" class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20211005)</code> was run prior to running the code in the R Markdown file. Setting a seed ensures that any results that rely on randomness, e.g. subsampling or permutations, are reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Session information:</strong> recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be confident that you successfully produced the results during this run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrong43be8a7"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Repository version:</strong> 43be8a7 </a>
</p>
</div>
<div id="strongRepositoryversionstrong43be8a7" class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development and connecting the code version to the results is critical for reproducibility.
</p>
<p>
The results in this page were generated with repository version 43be8a7. See the <em>Past versions</em> tab to see a history of the changes made to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for the analysis have been committed to Git prior to generating the results (you can use <code>wflow_publish</code> or <code>wflow_git_commit</code>). workflowr only checks the R Markdown file, but you know if there are other scripts or data files that it depends on. Below is the status of the Git repository when the results were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .DS_Store
    Ignored:    .Rhistory
    Ignored:    .Rproj.user/
    Ignored:    analysis/.DS_Store
    Ignored:    analysis/.Rhistory
    Ignored:    analysis/analysis_RNAseq_cache/
    Ignored:    data/.DS_Store
    Ignored:    output/.DS_Store

Untracked files:
    Untracked:  analysis/analysis_EMBL2016.Rmd
    Untracked:  analysis/analysis_GDSC.Rmd
    Untracked:  analysis/analysis_RNAseq.Rmd
    Untracked:  analysis/analysis_beatAML.Rmd
    Untracked:  analysis/process_EMBL2016.Rmd
    Untracked:  analysis/process_GDSC.Rmd
    Untracked:  analysis/process_beatAML.Rmd
    Untracked:  analysis/process_kinobeads.Rmd
    Untracked:  code/utils.R
    Untracked:  data/BeatAML/
    Untracked:  data/EMBL2016/
    Untracked:  data/GDSC/
    Untracked:  data/Kinobeads/
    Untracked:  data/RNAseq/
    Untracked:  manuscript/
    Untracked:  output/BeatAML_result.RData
    Untracked:  output/EMBL_result.RData
    Untracked:  output/EMBL_resultSub.RData
    Untracked:  output/GDSC_result.RData
    Untracked:  output/allTargets.rds
    Untracked:  output/inputs_BeatAML.RData
    Untracked:  output/inputs_EMBL.RData
    Untracked:  output/inputs_GDSC.RData

Unstaged changes:
    Modified:   README.md
    Modified:   _workflowr.yml
    Modified:   analysis/_site.yml
    Deleted:    analysis/about.Rmd
    Modified:   analysis/index.Rmd
    Deleted:    analysis/license.Rmd
    Deleted:    output/README.md

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not included in this status report because it is ok for generated content to have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">
<p>
There are no past versions. Publish this analysis with <code>wflow_publish()</code> to start tracking its development.
</p>
<hr>
</div>
</div>
</div>
<div id="introduction" class="section level1">
<h1>Introduction</h1>
<p>This document shows the preprocessing of the EMBL2016 screening dataset to use with the target importance inference package (DepInfeR) with the kinobeads kinase inhibitor screen (Klaeger, 2017).</p>
<hr />
</div>
<div id="load-packages" class="section level1">
<h1>Load packages</h1>
<p>Packages</p>
<pre class="r"><code>library(depInfeR)
library(stringdist)
library(BloodCancerMultiOmics2017)
library(DESeq2)
library(igraph)
library(tidyverse)
source(&quot;../code/utils.R&quot;)
knitr::opts_chunk$set(dev = c(&quot;png&quot;,&quot;pdf&quot;))</code></pre>
</div>
<div id="read-data-sets" class="section level1">
<h1>Read data sets</h1>
<p>Load pre-processed kinobead table table</p>
<pre class="r"><code>tarList &lt;- readRDS(&quot;../output/allTargets.rds&quot;)</code></pre>
<p>Read in EMBL2016 raw drug screen datasets</p>
<pre class="r"><code>EMBLscreen &lt;- readxl::read_xlsx(&quot;../data/EMBL2016/EMBL2016_screen.xlsx&quot;)

#sample annotation
patMeta &lt;- readxl::read_xlsx(&quot;../data/EMBL2016/EMBL2016_patAnnotation.xlsx&quot;)</code></pre>
</div>
<div id="preprocess-datasets" class="section level1">
<h1>Preprocess datasets</h1>
<div id="find-overlapping-drugs-between-drug-screen-data-and-drug-target-dataset" class="section level2">
<h2>Find overlapping drugs between drug screen data and drug-target dataset</h2>
<p>Get drug list from EMBL2016 screen</p>
<pre class="r"><code>drugList &lt;- EMBLscreen %&gt;% dplyr::select(drugID, name, Synonyms) %&gt;%
  filter(!is.na(drugID), !duplicated(drugID)) %&gt;% mutate(Drug = tolower(name)) %&gt;%
  mutate(Drug = gsub(&quot;[- ]&quot;,&quot;&quot;,Drug)) </code></pre>
<p>Find overlapped drugs by their names</p>
<pre class="r"><code>overDrug &lt;- intersect(tarList$Drug, drugList$Drug)</code></pre>
<p>Drugs that are not overlapped.</p>
<pre class="r"><code>missDrug &lt;- setdiff(drugList$Drug, tarList$Drug)</code></pre>
<p>Calculate hamming distance and consider synonyms</p>
<pre class="r"><code>notFound &lt;- setdiff(unique(tarList$Drug),overDrug)
stillNotFound &lt;- filter(drugList, Drug %in% missDrug)

distTab &lt;- lapply(seq(nrow(stillNotFound)), function(i) {
  drug1 &lt;- stillNotFound[i,]$Drug
  synList &lt;- strsplit(stillNotFound[i,]$Synonyms, split = &quot;,&quot;)[[1]]
  lapply(synList, function(syn) {
    lapply(notFound, function(drug2) {
      data.frame(drug1 = drug1, synonym = tolower(syn), drug2= drug2, dis = stringdist(tolower(syn), drug2), stringsAsFactors = FALSE)
    }) %&gt;% dplyr::bind_rows()
  }) %&gt;% dplyr::bind_rows()
}) %&gt;% dplyr::bind_rows()

distTab &lt;- arrange(distTab, dis)
head(distTab, n=10)</code></pre>
<pre><code>          drug1     synonym       drug2 dis
1   roscovitine  seliciclib  seliciclib   0
2  flavopiridol   alvocidib   alvocidib   0
3     nvpaew541      aew541      aew541   0
4       azd9291 osimertinib osimertinib   0
5   afuresertib  gsk2110183  gsk2110183   0
6        sns032  bms-387032   bms387032   1
7        mk8776  sch 900776   sch900776   1
8        bi6727  volasertib  volasertib   1
9   roscovitine  seliciclib   milciclib   3
10      azd9291 osimertinib ulixertinib   3</code></pre>
<p>The first 8 drugs are the same drugs</p>
<p>Get drug mappings</p>
<pre class="r"><code>drugMap &lt;- distTab[1:8,]$drug1
names(drugMap) &lt;- distTab[1:8,]$drug2</code></pre>
<p>Modify the name</p>
<pre class="r"><code>tarList &lt;- mutate(tarList, Drug = ifelse(Drug %in% names(drugMap), drugMap[Drug],Drug))</code></pre>
<p>Get the final overlapped drug list</p>
<pre class="r"><code>finalList &lt;- intersect(tarList$Drug, drugList$Drug)</code></pre>
<p>Combine the lists and match drug IDs</p>
<pre class="r"><code>targets &lt;- left_join(tarList, drugList, by = &quot;Drug&quot;) %&gt;% 
  dplyr::select(name, drugID, `Target Classification`, EC50,`Apparent Kd`, `Gene Name`) %&gt;%
  dplyr::filter(!is.na(name))</code></pre>
<p>How many drugs?</p>
<pre class="r"><code>length(unique(targets$drugID))</code></pre>
<pre><code>[1] 86</code></pre>
<p>Change names</p>
<pre class="r"><code>colnames(targets) &lt;- c(&quot;drugName&quot;,&quot;drugID&quot;,&quot;targetClassification&quot;,&quot;EC50&quot;,&quot;Kd&quot;,&quot;targetName&quot;,&quot;originalTarget&quot;,&quot;originalPathway&quot;)</code></pre>
</div>
<div id="remove-targets-that-are-not-expressed-in-patient-samples-at-all" class="section level2">
<h2>Remove targets that are not expressed in patient samples at all</h2>
<p>Based on published RNAseq dataset</p>
<pre class="r"><code>data(&quot;dds&quot;)
dds &lt;- dds[,dds$PatID %in% EMBLscreen$patID]
colnames(dds) &lt;- dds$PatID</code></pre>
<p>Get count values from RNAseq data</p>
<pre class="r"><code>#targets that are not in RNAseq dataset
#setdiff(unique(targets$targetName), rowData(dds)$symbol)

#actually four genes have different gene names used.
symbolMap &lt;- c(ADCK3 =&quot;COQ8A&quot;, ZAK = &quot;MAP3K20&quot;,
               KIAA0195 = &quot;TMEM94&quot;, ADRBK1 = &quot;GRK2&quot;)

#correct the name
targets &lt;- mutate(targets, targetName = ifelse(targetName %in% names(symbolMap),
                                  symbolMap[targetName],
                                  targetName))
highTargets &lt;- filter(targets, targetClassification == &quot;High confidence&quot;)

#get count data
targetCount &lt;- dds[rowData(dds)$symbol %in% targets$targetName,]</code></pre>
<p>Plot the expression values</p>
<pre class="r"><code>#prepare plot tab
plotTab &lt;- data.frame(counts(targetCount, normalized = FALSE)) %&gt;% 
  rownames_to_column(&quot;ID&quot;) %&gt;%
  mutate(symbol = rowData(targetCount)$symbol) %&gt;%
  gather(key = &quot;patID&quot;, value = &quot;counts&quot;, -symbol, -ID)

#deal with one gene, multiple transcript problem
#only keep the most aboundant transcript
transTab &lt;- group_by(plotTab, ID, symbol) %&gt;% summarize(total = sum(counts)) %&gt;%
  ungroup() %&gt;%
  arrange(desc(total)) %&gt;% distinct(symbol, .keep_all = TRUE)</code></pre>
<pre><code>`summarise()` has grouped output by &#39;ID&#39;. You can override using the `.groups` argument.</code></pre>
<pre class="r"><code>plotTab &lt;- filter(plotTab, ID %in% transTab$ID)


#get the 80% quantile expression value
exprMed &lt;- group_by(plotTab, symbol) %&gt;% summarise(avgCount = quantile(counts,0.8)) %&gt;%
   arrange(avgCount) %&gt;% top_n(-50, avgCount)

#only plot the 50 lowest expressed genes
plotTab &lt;- filter(plotTab, symbol %in% exprMed$symbol) %&gt;%
  mutate(symbol = factor(symbol, levels = exprMed$symbol))

ggplot(plotTab, aes(x= symbol, y = counts)) + geom_boxplot() +
  theme_bw() + theme(axis.text.x = element_text(angle = 90, hjust =1, vjust =.5)) </code></pre>
<p><img src="figure/process_EMBL2016.Rmd/unnamed-chunk-16-1.png" width="960" style="display: block; margin: auto;" /></p>
<p>Removed the targets that are not expressed</p>
<pre class="r"><code>#80% quantile &lt; 10
geneRemove &lt;- filter(exprMed, rank(avgCount) / n() &lt; 0.8)
geneRemove &lt;- filter(exprMed,avgCount &lt; 10)$symbol
targets &lt;- filter(targets, !targetName %in% geneRemove)</code></pre>
<p>Turn target table into drug-target affinity matrix</p>
<pre class="r"><code>tarMat_kd &lt;- dplyr::filter(targets, targetClassification == &quot;High confidence&quot;) %&gt;% 
    dplyr::select(drugID, targetName, Kd) %&gt;% 
    spread(key = &quot;targetName&quot;, value = &quot;Kd&quot;) %&gt;%
    remove_rownames() %&gt;% column_to_rownames(&quot;drugID&quot;) %&gt;% as.matrix()</code></pre>
</div>
<div id="re-scale-kd-values-and-collapse-highly-correlated-drugs" class="section level2">
<h2>Re-scale Kd values and collapse highly correlated drugs</h2>
<p>Apply drug-target preprocessing function</p>
<pre class="r"><code>ProcessTargetResults &lt;- processTarget(tarMat_kd, ifKd = TRUE , ifCorFeat = TRUE, keepTargets = c(&quot;BTK&quot;,&quot;LYN&quot;,&quot;SRC&quot;,&quot;SYK&quot;, &quot;WEE1&quot;))</code></pre>
</div>
<div id="plot-target-groups" class="section level2">
<h2>Plot target groups</h2>
<pre class="r"><code>#plot network
#Only plot for finnally selected targets

load(&quot;../output/EMBL_result.RData&quot;)
CancerxTargets&lt;- rowSums(result$freqMat)
CancerxTargets &lt;- names(CancerxTargets[CancerxTargets&gt;0])
#CancerxTargets &lt;- c(&quot;MAP2K2&quot;, &quot;MAP4K2&quot;, &quot;BTK&quot;, &quot;PTK6&quot;, &quot;WEE1&quot;, &quot;CHEK1&quot;)
plotTarGroups(ProcessTargetResults, CancerxTargets)</code></pre>
<p><img src="figure/process_EMBL2016.Rmd/targetNet_EMBL2016-1.png" width="384" style="display: block; margin: auto;" /></p>
</div>
<div id="visualization-of-the-whole-drug-target-network" class="section level2">
<h2>Visualization of the whole drug-target network</h2>
<pre class="r"><code>plotTab &lt;- dplyr::select(targets, drugName, targetName)
nodeAttr &lt;- gather(plotTab, key = &quot;type&quot;, value = &quot;name&quot;, drugName, targetName) %&gt;%
  filter(!duplicated(name)) %&gt;%
  mutate(type = ifelse(type == &quot;targetName&quot;, &quot;target&quot;, &quot;drug&quot;))

g &lt;- graph_from_edgelist(as.matrix(plotTab))

V(g)$nodeType &lt;- nodeAttr[match(V(g)$name, nodeAttr$name),]$type
V(g)$shape &lt;- ifelse(V(g)$nodeType == &quot;drug&quot;, &quot;circle&quot;,&quot;square&quot;)
V(g)$color &lt;- ifelse(V(g)$nodeType == &quot;drug&quot;, &quot;skyblue&quot;,&quot;pink&quot;)
V(g)$size = 6
V(g)$label.cex = 0.7
plot(g, layout=layout_with_kk)</code></pre>
<p><img src="figure/process_EMBL2016.Rmd/drugTargetNet_EMBL-1.png" width="1920" style="display: block; margin: auto;" /> No obvious structure can be seen. Polypharmacology needs to be resolved.</p>
</div>
<div id="preparation-of-drug-response-matrix" class="section level2">
<h2>Preparation of drug response matrix</h2>
<div id="prepare-response-matrix-using-the-z-score" class="section level3">
<h3>Prepare response matrix using the z-score</h3>
<p>In order to be consistent for all drugs, only the 9 lowest concentrations are regarded.</p>
<p><strong>Use average of 9 concentrations</strong></p>
<pre class="r"><code>viabTab &lt;- dplyr::filter(EMBLscreen,
                  concIndex %in% seq(1,9)) %&gt;% 
  group_by(drugID, patID) %&gt;% 
  summarise(viab = mean(normVal.sigm)) %&gt;% ungroup() %&gt;%
  dplyr::rename(Drug = drugID, patientID = patID)</code></pre>
<pre><code>`summarise()` has grouped output by &#39;drugID&#39;. You can override using the `.groups` argument.</code></pre>
<pre class="r"><code>viabMat &lt;- spread(viabTab, patientID, viab) %&gt;%
  data.frame() %&gt;%
  column_to_rownames(&quot;Drug&quot;) %&gt;% as.matrix()</code></pre>
</div>
</div>
</div>
<div id="save-pre-processed-dataset" class="section level1">
<h1>Save pre-processed dataset</h1>
<pre class="r"><code>targetsEMBL &lt;- targets
ProcessTargetResults_EMBL &lt;- ProcessTargetResults
tarMat_EMBL &lt;- ProcessTargetResults$resTarMat
viabMat_EMBL &lt;- viabMat[rownames(tarMat_EMBL),]
annotation_EMBL &lt;- patMeta
save(tarMat_EMBL, viabMat_EMBL, annotation_EMBL, ProcessTargetResults_EMBL, targetsEMBL, file = &quot;../output/inputs_EMBL.RData&quot;)</code></pre>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span> Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 4.0.2 (2020-06-22)
Platform: x86_64-apple-darwin17.0 (64-bit)
Running under: macOS  10.16

Matrix products: default
BLAS:   /Library/Frameworks/R.framework/Versions/4.0/Resources/lib/libRblas.dylib
LAPACK: /Library/Frameworks/R.framework/Versions/4.0/Resources/lib/libRlapack.dylib

locale:
[1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8

attached base packages:
[1] parallel  stats4    stats     graphics  grDevices utils     datasets 
[8] methods   base     

other attached packages:
 [1] forcats_0.5.1                   stringr_1.4.0                  
 [3] dplyr_1.0.7                     purrr_0.3.4                    
 [5] readr_2.0.0                     tidyr_1.1.3                    
 [7] tibble_3.1.3                    ggplot2_3.3.5                  
 [9] tidyverse_1.3.1                 igraph_1.2.6                   
[11] DESeq2_1.28.1                   SummarizedExperiment_1.18.2    
[13] DelayedArray_0.14.1             matrixStats_0.60.0             
[15] Biobase_2.48.0                  GenomicRanges_1.40.0           
[17] GenomeInfoDb_1.24.2             IRanges_2.22.2                 
[19] S4Vectors_0.26.1                BiocGenerics_0.34.0            
[21] BloodCancerMultiOmics2017_1.8.0 stringdist_0.9.7               
[23] depInfeR_0.1.0                 

loaded via a namespace (and not attached):
  [1] readxl_1.3.1           backports_1.2.1        Hmisc_4.5-0           
  [4] workflowr_1.6.2        plyr_1.8.6             splines_4.0.2         
  [7] BiocParallel_1.22.0    usethis_2.0.1          digest_0.6.27         
 [10] foreach_1.5.1          htmltools_0.5.1.1      fansi_0.5.0           
 [13] checkmate_2.0.0        magrittr_2.0.1         memoise_2.0.0         
 [16] cluster_2.1.2          doParallel_1.0.16      tzdb_0.1.2            
 [19] remotes_2.4.0          annotate_1.66.0        modelr_0.1.8          
 [22] prettyunits_1.1.1      jpeg_0.1-9             colorspace_2.0-2      
 [25] blob_1.2.2             rvest_1.0.1            haven_2.4.1           
 [28] xfun_0.24              callr_3.7.0            crayon_1.4.1          
 [31] RCurl_1.98-1.3         jsonlite_1.7.2         genefilter_1.70.0     
 [34] survival_3.2-11        iterators_1.0.13       glue_1.4.2            
 [37] gtable_0.3.0           zlibbioc_1.34.0        XVector_0.28.0        
 [40] pkgbuild_1.2.0         shape_1.4.6            abind_1.4-5           
 [43] scales_1.1.1           DBI_1.1.1              Rcpp_1.0.7            
 [46] htmlTable_2.2.1        xtable_1.8-4           tmvnsim_1.0-2         
 [49] foreign_0.8-81         bit_4.0.4              Formula_1.2-4         
 [52] ipflasso_1.1           glmnet_4.1-2           htmlwidgets_1.5.3     
 [55] httr_1.4.2             lavaan_0.6-9           RColorBrewer_1.1-2    
 [58] ellipsis_0.3.2         pkgconfig_2.0.3        XML_3.99-0.6          
 [61] farver_2.1.0           nnet_7.3-16            sass_0.4.0            
 [64] dbplyr_2.1.1           locfit_1.5-9.4         utf8_1.2.2            
 [67] tidyselect_1.1.1       labeling_0.4.2         rlang_0.4.11          
 [70] reshape2_1.4.4         later_1.2.0            AnnotationDbi_1.50.3  
 [73] munsell_0.5.0          cellranger_1.1.0       tools_4.0.2           
 [76] cachem_1.0.5           cli_3.0.1              generics_0.1.0        
 [79] RSQLite_2.2.7          devtools_2.4.2         broom_0.7.9           
 [82] fdrtool_1.2.16         evaluate_0.14          fastmap_1.1.0         
 [85] ggdendro_0.1.22        yaml_2.2.1             processx_3.5.2        
 [88] knitr_1.33             bit64_4.0.5            fs_1.5.0              
 [91] glasso_1.11            pbapply_1.4-3          nlme_3.1-152          
 [94] xml2_1.3.2             compiler_4.0.2         rstudioapi_0.13       
 [97] png_0.1-7              beeswarm_0.4.0         testthat_3.0.4        
[100] reprex_2.0.0           geneplotter_1.66.0     pbivnorm_0.6.0        
[103] bslib_0.2.5.1          stringi_1.7.3          highr_0.9             
[106] ps_1.6.0               qgraph_1.6.9           desc_1.3.0            
[109] lattice_0.20-44        Matrix_1.3-4           psych_2.1.6           
[112] ggsci_2.9              vctrs_0.3.8            pillar_1.6.2          
[115] lifecycle_1.0.0        jquerylib_0.1.4        corpcor_1.6.9         
[118] data.table_1.14.0      bitops_1.0-7           httpuv_1.6.1          
[121] latticeExtra_0.6-29    R6_2.5.0               promises_1.2.0.1      
[124] gridExtra_2.3          sessioninfo_1.1.1      codetools_0.2-18      
[127] gtools_3.9.2           MASS_7.3-54            assertthat_0.2.1      
[130] pkgload_1.2.1          rprojroot_2.0.2        withr_2.4.2           
[133] mnormt_2.0.2           GenomeInfoDbData_1.2.3 rlist_0.4.6.1         
[136] hms_1.1.0              rpart_4.1-15           grid_4.0.2            
[139] rmarkdown_2.9          git2r_0.28.0           base64enc_0.1-3       
[142] lubridate_1.7.10      </code></pre>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
http://docs.mathjax.org/en/latest/configuration.html.  This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>




</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
